File: iOS/TUIKit/TUIRoomKit/Source/ConferenceSession.swift
//
//  ConferenceSession.swift
//  TUIRoomKit
//
//  Created by CY zhao on 2024/8/9.
//

import Foundation

@objcMembers public class ConferenceSession: NSObject {
    @objc public static let sharedInstance = ConferenceSession()
    let implementation = ConferenceSessionImp()
    
    @objc public func addObserver(observer: ConferenceObserver) {
        implementation.addObserver(observer: observer)
    }
    
    @objc public func removeObserver(observer: ConferenceObserver) {
        implementation.removeObserver(observer: observer)
    }
    
    @objc public static func destroySharedInstance() {
        sharedInstance.implementation.destroy()
    }
    
    @objc public func enableWaterMark() {
        implementation.enableWaterMark()
    }
    
    @objc public func setWaterMarkText(waterMarkText: String) {
        implementation.setWaterMarkText(waterMarkText: waterMarkText)
    }
    
    @objc public func setContactsViewProvider(_ provider: @escaping (ConferenceParticipants) -> ContactViewProtocol) {
        implementation.setContactsViewProvider(provider)
    }
        
}

File: Android/TUIKit/TUIRoomKit/tuiroomkit/src/main/java/com/tencent/cloud/tuikit/roomkit/ConferenceSession.java
package com.tencent.cloud.tuikit.roomkit;

import com.tencent.cloud.tuikit.roomkit.impl.ConferenceSessionImpl;

import java.util.List;

public abstract class ConferenceSession {
    public static ConferenceSession sharedInstance() {
        return ConferenceSessionImpl.sharedInstance();
    }

    public static void destroySharedInstance() {
        ConferenceSessionImpl.destroySharedInstance();
    }

    public abstract void addObserver(ConferenceDefine.ConferenceObserver observer);

    public abstract void removeObserver(ConferenceDefine.ConferenceObserver observer);

    public abstract void enableWaterMark();

    public abstract void setWaterMarkText(String waterMarkText);

    public abstract void setContactsViewProvider(Class<?> contactsActivity);

    public abstract void setParticipants(List<ConferenceDefine.User> participants);
}
